content/React Query and management of server state.md

```diff
@@ -0,0 +1,75 @@
+# React Query and management of server state
+
+Some days ago the [React Summit Remote Edition](https://remote.reactsummit.com/) gathered tons of people behind the screen with a promise of put together popular speakers around the React Community like [Kent C. Dodds](https://kentcdodds.com/) and [Guillermo Rauch](https://rauchg.com/) among others, but one talk especially caught my attention, *React Query: It’s Time to Break up with your "Global State”!* by [Tanner Linsley](https://twitter.com/tannerlinsley) were joining two pain points that I had frequently with React, such as how to handle asynchronous information in the global state and how to restructure my code to be less repetitive and not force async data to follow synchronous flow.
+In this article, I explain more in deep which is the problem with the global state and how React Query solves it in an efficient-scalable way. 
+
+## The problem with global state
+Libraries like Redux, MobX and even Context API provided by React are based in the Global State pattern, which means that to avoid prop drilling between the components and have one single point to read and write from the components, the entire application exposes a global variable called _global state_.
+
+This pattern has many advantages as I mentioned before, but the problem **lies in the different types of data that potentially have the global state in our applications**.
+These data can be internal information about our application, like for example if the navbar is open or not, or information that has ownership with another source, for instance, the user information like his name, age, etc, that is provided by backend and can be modified in the frontend.
+
+These two types of data can be classified as a **client state and server state**.
+The main difference between each other is the server state has two sources of truth, the backend and the frontend. This causes that the server state has to synchronize all the time to prevent that his information is not out of date. As we know, is not performant at all and is not desired by our users(image put a spinner every time you do a request) request information all the time, for that reason **we need in some way show the old information and at the same time updating with the new information on the background every so often**. 
+This is a very difficult task if we try to do with Redux or the other tools that I mentioned before because we have to find a way to resolve cache management, background update and other cases that precise a little bit more of implementation when merging old and new data such as pagination or infinite scroll.
+
+## What is React Query and how it solves these problems?
+React query is a library that with 2 hooks and 1 utility (only 5kb!) provides an easy and scalable way to fetching, caching and updating asynchronous data in React.
+
+Before jump off to the code and see some examples I like to introduce how React Query models the server state.
```

Comment:
`Before _jumping_ off to the code and _seeing_ some examples `I'd` like to introduce how React Query models the server state.`